Source/WebCore/Modules/filesystemaccess/FileSystemSyncAccessHandle.cpp

```diff
@@ -55,161 +55,81 @@ FileSystemSyncAccessHandle::FileSystemSyncAccessHandle(ScriptExecutionContext& c
 
 FileSystemSyncAccessHandle::~FileSystemSyncAccessHandle()
 {
-    ASSERT(isClosingOrClosed());
-
     m_source->unregisterSyncAccessHandle(m_identifier);
-
-    if (m_closeResult)
-        return;
-
-    // Task replies may be scheduled to WorkerRunLoop but do not run.
-    auto pendingPromises = std::exchange(m_pendingPromises, { });
-    for (auto& promise : pendingPromises) {
-        std::visit([](auto& promise) {
-            promise.reject(Exception { UnknownError, "AccessHandle is about to be destroyed"_s });
-        }, promise);
-    }
-
-    closeBackend(CloseMode::Sync);
+    closeInternal(ShouldNotifyBackend::Yes);
 }
 
-bool FileSystemSyncAccessHandle::isClosingOrClosed() const
+ExceptionOr<void> FileSystemSyncAccessHandle::truncate(unsigned long long size)
 {
-    return m_closeResult || !m_closeCallbacks.isEmpty();
-}
-
-void FileSystemSyncAccessHandle::truncate(unsigned long long size, DOMPromiseDeferred<void>&& promise)
-{
-    if (isClosingOrClosed())
-        return promise.reject(Exception { InvalidStateError, "AccessHandle is closing or closed"_s });
+    if (m_isClosed)
+        return Exception { InvalidStateError, "AccessHandle is closed"_s };
 
     auto* scope = downcast<WorkerGlobalScope>(scriptExecutionContext());
     if (!scope)
```

Comment:
Pre-existing issue but we do not use scope, do we need the downcast and the if check (here and below)?

Reply:
Yes, we probably should remove it now that they are unused; will do.